src/neo/Ledger/Blockchain.cs

```diff
@@ -367,6 +367,7 @@ private VerifyResult OnNewBlock(Block block)
                         Self.Tell(unverifiedBlock, ActorRefs.NoSender);
                     block_cache_unverified.Remove(Height + 1);
                 }
+                block_cache.Add(block.Hash, block);
```

Comment:
`Add` could throw an error if it's already in the dictionary, maybe we should try to avoid it.

```suggestion
                block_cache[block.Hash] = block;
```

Reply:
Fixed. There are also many other places where similiarly Dictionary.Add used, maybe we can start a new PR correcting them.